icfnext/storybook-aem
# Java proxy servlet

View the aem-storybook-core README for how to configure and use the proxy. Right now this needs manually installed with maven but I will look into ways of installing it automatically as just one more option like any of the others.

### TODO
- [x] Create a storybook-aem-foundation package.
- [x] Create the proxy servlet.
- [x] Add cli install prompts for the packages.
- [x] Auto package install. This has been started and a TODO is in the code.
- [x] Test the proxy servlet.

### Testing the install
1. Run the `storybook-aem init` command to initialize the project.
1. When prompted say yes to the Confluence optional package
1. When prompted, use the default AEM port of `4502`.
1. After the install has finished go to /system/console/configMgr in AEM and search for "Storybook AEM Proxy" and ensure that this configuration exists.

### Testing the proxy
1. Have storybook-aem properly configured and running in AEM with the Confluence package in use.
1. Go to /system/console/configMgr in AEM and search for "Storybook AEM Proxy" and configure the necessary authentication headers for Confluence.
1. Go to Storybook and ensure that the Confluence package can reach the Confluence server.

CC: @jzeltman
